File: work/deprecated/room/entity/HistoryEntity.java
package wangdaye.com.geometricweather.room.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.Date;

import wangdaye.com.geometricweather.basic.models.options.provider.WeatherSource;

/**
 * History entity.
 *
 * {@link wangdaye.com.geometricweather.basic.models.weather.History}.
 * */

@Entity(tableName = "HISTORY_ENTITY")
public class HistoryEntity {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id") public Long id;
    @ColumnInfo(name = "CITY_ID") public String cityId;
    @ColumnInfo(name = "WEATHER_SOURCE") public WeatherSource weatherSource;

    @ColumnInfo(name = "DATE") public Date date;
    @ColumnInfo(name = "TIME") public long time;

    @ColumnInfo(name = "DAYTIME_TEMPERATURE") public int daytimeTemperature;
    @ColumnInfo(name = "NIGHTTIME_TEMPERATURE") public int nighttimeTemperature;
}

File: app/src/main/java/wangdaye/com/geometricweather/db/entities/HistoryEntity.kt
package wangdaye.com.geometricweather.db.entities

import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id
import java.util.Date

/**
 * History entity.
 *
 * [History].
 */
@Entity
data class HistoryEntity(
    @field:Id var id: Long = 0,
    var cityId: String,
    var weatherSource: String,
    var date: Date,
    var daytimeTemperature: Int? = null,
    var nighttimeTemperature: Int? = null
)
